cmd/kubectl-k8ssandra/register/registration.go

```diff
@@ -0,0 +1,143 @@
+package register
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/k8ssandra/k8ssandra-client/pkg/registration"
+	configapi "github.com/k8ssandra/k8ssandra-operator/apis/config/v1beta1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/result"
+)
+
+type RegistrationExecutor struct {
+	DestinationName  string
+	SourceKubeconfig string
+	DestKubeconfig   string
+	SourceContext    string
+	DestContext      string
+	SourceNamespace  string
+	DestNamespace    string
+	ServiceAccount   string
+	Context          context.Context
+}
+
+func getDefaultSecret(saName, saNamespace string) *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      saName + "-secret",
+			Namespace: saNamespace,
+			Annotations: map[string]string{
+				"kubernetes.io/service-account.name": saName,
+			},
+		},
+		Type: corev1.SecretTypeServiceAccountToken,
+	}
+}
+func getDefaultServiceAccount(saName, saNamespace string) *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      saName,
+			Namespace: saNamespace,
+		},
+	}
+
+}
+
+func (e *RegistrationExecutor) RegisterCluster() result.ReconcileResult {
+	log.Printf("Registering cluster from %s Context: %s to %s Context: %s",
+		registration.GetKubeconfigFileLocation(e.SourceKubeconfig), e.SourceContext,
+		registration.GetKubeconfigFileLocation(e.DestKubeconfig), e.DestContext,
+	)
+	if e.SourceContext == e.DestContext && e.SourceKubeconfig == e.DestKubeconfig {
+		panic(errors.New("source and destination context and kubeconfig are the same, you should not register the same cluster to itself. Reference it by leaving the k8sContext field blank instead"))
+	}
+	srcClient, err := registration.GetClient(e.SourceKubeconfig, e.SourceContext)
+	if err != nil {
+		return result.Error(err)
+	}
+	destClient, err := registration.GetClient(e.DestKubeconfig, e.DestContext)
+	if err != nil {
+		return result.Error(err)
+	}
+	// Get ServiceAccount
+	serviceAccount := &corev1.ServiceAccount{}
+	if err := srcClient.Get(e.Context, client.ObjectKey{Name: e.ServiceAccount, Namespace: e.SourceNamespace}, serviceAccount); err != nil {
+		if err := srcClient.Create(e.Context, getDefaultServiceAccount(e.ServiceAccount, e.SourceNamespace)); err != nil {
```

Comment:
issue: We should not call Create if we get an error. Only if the error is of type NotFound, otherwise we might have errors such as connection error to the server.

Reply:
Indeed, thanks for catching this.